vmaas_sync/advisory_sync.go

```diff
@@ -0,0 +1,140 @@
+package vmaas_sync
+
+import (
+	"app/base"
+	"app/base/database"
+	"app/base/models"
+	"app/base/utils"
+	"context"
+	"github.com/RedHatInsights/patchman-clients/vmaas"
+	"github.com/antihax/optional"
+	"github.com/pkg/errors"
+	"time"
+)
+
+
+// Should be < 5000
+const SYNC_BATCH_SIZE = 1000
+
+var (
+	vmaasClient *vmaas.APIClient
+)
+
+func configure() {
+	cfg := vmaas.NewConfiguration()
+	cfg.BasePath = utils.GetenvOrFail("VMAAS_ADDRESS") + base.VMAAS_API_PREFIX
+	cfg.Debug = true
+
+	vmaasClient = vmaas.NewAPIClient(cfg)
+}
+
+func parseAdvisories(data map[string]vmaas.ErrataResponseErrataList) (models.AdvisoryMetadataSlice, error) {
+	var advisories models.AdvisoryMetadataSlice
+
+	// We use advisory types from DB
+	var advisoryTypesArr []models.AdvisoryType
+	advisoryTypes := map[string]int{}
+
+	err := database.Db.Find(&advisoryTypesArr).Error
+	if err != nil {
+		return nil, errors.WithMessage(err, "Loading advisory types")
+	}
+
+	for _, t := range advisoryTypesArr {
+		advisoryTypes[t.Name] = t.ID
+	}
+
+	for n, v := range data {
+		// TODO: Should we skip or report invalid erratas ?
+		issued, err := time.Parse(base.RFC_3339_NO_TZ, v.Issued)
+		if err != nil {
+			utils.Log("err", err.Error()).Error("Invalid errata issued date")
```

Comment:
I get this error when testing:
```
vmaas_sync_1  | time="2020-01-06T14:21:41Z" level=debug msg="Downloaded advisories" count=3
vmaas_sync_1  | time="2020-01-06T14:21:41Z" level=error msg="Invalid errata issued date" err="parsing time \"2006-13-55T15:04:05+07:00\": month out of range"
```

Can the errata ID be added to the error? This would make debugging easier in production

Reply:
Added, thanks for suggestion.